app/models/case/base.rb

```diff
@@ -962,5 +974,21 @@ def validate_case_link(type, linked_case, attribute)
     end
   end
 
+  def find_timing_for_extending_deadline
+    if potential_include_extension?
+      most_recent_extension = self.transitions
+      .where(event: ['extend_for_pit', 'extend_sar_deadline'])
+      .order(id: :desc)
+      .first
+      most_recent_extension.nil? ? nil : most_recent_extension.created_at.to_date
+    else
+      nil
+    end
+  end
+
+  def potential_include_extension?
```

Comment:
I find this method name a bit opaque, not clear what the concept is here :) - wondering if we can consider changing it to make it a bit more self-documenting?

Reply:
I am thinking I will remove this func and only check the flag of has_pit_extension as this is London user needs,  I was thinking maybe it is nice to check 'SAR' case too---> now I feel I shouldn't really think about SAR case type as it maybe be managed differently.